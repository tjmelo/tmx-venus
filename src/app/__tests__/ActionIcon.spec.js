import toActionIcon from '../modules/ActionIcon';

const testArgs = {
  testButtonElement: 'button',
  testIconReference: '<i id="icon-test" title="Test" class="d-none fas fa-cart-arrow-test"></i>',
};

document.body.innerHTML = `<${testArgs.testButtonElement}>${testArgs.testIconReference}</${testArgs.testButtonElement}>
`;

describe('Should verify if the toActionIcon function has been called', () => {
  const testElementToBeCompared = document.querySelector('button');

  it('Should to call the function', () => {
    const testActionIcon = jest
      .fn()
      .mockImplementationOnce((argTest1, argTest2) => toActionIcon(argTest1, argTest2));

    const { testButtonElement, testIconReference } = testArgs;

    // do
    testActionIcon(testButtonElement, testIconReference);

    // then
    expect(testActionIcon).toHaveReturned();
    expect(testActionIcon).toHaveBeenCalledTimes(1);
    expect(testActionIcon).toHaveBeenCalledWith(testButtonElement, testIconReference);

    expect(testActionIcon).toHaveReturnedWith(testElementToBeCompared);

    // Expects fidelity with scope of function
    expect(testActionIcon).toMatchSnapshot();
  });
});
