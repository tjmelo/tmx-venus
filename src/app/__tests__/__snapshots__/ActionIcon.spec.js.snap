// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should verify if the toActionIcon function has been called Should to call the function 1`] = `
[MockFunction] {
  "calls": [
    [
      "button",
      "<i id="icon-test" title="Test" class="d-none fas fa-cart-arrow-test"></i>",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": <button>
        <i
          class="d-none fas fa-cart-arrow-test"
          id="icon-test"
          title="Test"
        />
      </button>,
    },
  ],
}
`;
